src/renderer/mixins/addRepository.js

```diff
@@ -66,8 +67,16 @@ export default {
 
 		// validate git repository
 		async isGitRepo(path) {
-			let repository = await nodegit.Repository.open(path);
 			try {
+				let repository;
+				try {
+					repository = await nodegit.Repository.open(path);
+				} catch (_) {}
+				let autoInitFeature = this.$store.getters["settings/getExperimental"]
+					.autoInit;
+				if (!repository && autoInitFeature) {
+					gitInit({ path });
+				}
 				this.newRepository.isGitRepo = !!repository;
```

Comment:
```suggestion
				this.newRepository.isGitRepo = true;
```

Reply:
I tried few ways to get the `Repository {}` object, seems like it not working. So for now we go with hard-coded true value.

Not the best option, I know, but I will look into improving the gitInit API further.